rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -664,7 +664,8 @@ get_info_by_topic(
   rclcpp::node_interfaces::NodeBaseInterface * node_base,
   const std::string & topic_name,
   bool no_mangle,
-  FunctionT rcl_get_info_by_topic)
+  bool is_service,
+  FunctionT rcl_get_info_by_topic_or_service)
```

Comment:
can we extend this function to support endpoint based on `EndpointTypes` with topics, services (and actions in the future?) there is already the template based on `EndpointTypes` so extend `EndpointTypes` with service types makes this function more generic like `get_info_by_endpoint`?